From 4006783a6391d424181c62426ab08b9ae1714240 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Fri, 26 Jul 2024 17:15:09 +0800
Subject: [PATCH 31/31] Make more deps entries public_deps in
 chrome/browser/ui/BUILD.gn

The chrome/browser/ui build target includes several buildflag headers.
For this to work correctly, the build system must be able to generate
the buildflag headers first.  To do this reliably, without having to
depend on buildflag targets in all targets that transitively include the
headers that includes buildflag heaers, make these dependencies
public_deps.

For //components/paint_preview/buildflags, remove the deps entry, as it
is already in public_deps.

Bug: 40253918, 346711540, 346699817, 345645751
Change-Id: I00e78a430f2e99aa3732406882b352d31e3d7da7
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5647662
Commit-Queue: Lei Zhang <thestig@chromium.org>
Reviewed-by: Peter Kasting <pkasting@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1320033}

Co-authored-by: Lei Zhang <thestig@chromium.org>
---
 chrome/browser/ui/BUILD.gn | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/chrome/browser/ui/BUILD.gn b/chrome/browser/ui/BUILD.gn
index 3d1ccee5c6..738bed56e8 100644
--- a/chrome/browser/ui/BUILD.gn
+++ b/chrome/browser/ui/BUILD.gn
@@ -422,8 +422,10 @@ static_library("ui") {
     "//chrome/browser/ui/tabs:tab_enums",
     "//components/cross_device/logging",
     "//components/dom_distiller/core",
+    "//components/enterprise/buildflags",
     "//components/paint_preview/buildflags",
     "//components/safe_browsing:buildflags",
+    "//components/segmentation_platform/public",
     "//components/sync",
     "//components/sync_user_events",
     "//components/translate/content/browser",
@@ -540,7 +542,6 @@ static_library("ui") {
     "//components/embedder_support:browser_util",
     "//components/encrypted_messages:encrypted_message_proto",
     "//components/enterprise",
-    "//components/enterprise/buildflags:buildflags",
     "//components/error_page/content/browser",
     "//components/facilitated_payments/core/features",
     "//components/favicon/content",
@@ -651,7 +652,6 @@ static_library("ui") {
     "//components/security_interstitials/core:unsafe_resource",
     "//components/security_state/content",
     "//components/security_state/core",
-    "//components/segmentation_platform/public",
     "//components/send_tab_to_self",
     "//components/sessions",
     "//components/signin/core/browser",
@@ -2181,7 +2181,6 @@ static_library("ui") {
       "//components/omnibox/browser:mojo_bindings",
       "//components/page_load_metrics/browser",
       "//components/paint_preview/browser",
-      "//components/paint_preview/buildflags",
       "//components/paint_preview/common",
       "//components/paint_preview/public",
       "//components/password_manager/content/common",
-- 
2.45.2

